toolchain/check/impl_lookup.cpp

```diff
@@ -1009,94 +1008,71 @@ auto EvalLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
   // not be concrete in this case, so only final impls can produce a concrete
   // witness for this query.
   auto candidates = CollectCandidateImplsForQuery(
-      context, kind == FoundInFacet, query_self_const_id, *query_type_structure,
-      query_specific_interface);
+      context, facet_lookup_result.has_value(), query_self_const_id,
+      *query_type_structure, query_specific_interface);
+
+  bool query_is_concrete =
+      QueryIsConcrete(context, query_self_const_id, query_specific_interface);
+  CARBON_CHECK(!query_is_concrete || !facet_lookup_result.has_value(),
+               "Non-concrete facet lookup value for concrete query");
+
+  // Perform a lookup for an `impl` that matches the query. If we don't find a
+  // final impl, the self value may still have been a facet that provides a
+  // symbolic witness in the `facet_lookup_result`, which we want to fall back
+  // to. It records that an `impl` will exist for the query, but is yet unknown.
+
+  struct LookupResult {
+    EvalImplLookupResult result;
+    std::optional<TypeStructure> impl_type_structure;
+    SemIR::LocId impl_loc = SemIR::LocId::None;
+  };
+
+  auto lookup_result = LookupResult{.result = facet_lookup_result};
```

Comment:
```suggestion
  LookupResult lookup_result = {.result = facet_lookup_result};
```

Initialization style nit: the preference is to avoid `type{...}` if possible, including as suggested

Reply:
Done